src/OrbitGl/include/OrbitGl/CallTreeView.h

```diff
@@ -188,20 +188,42 @@ class CallTreeUnwindErrorType : public CallTreeNode {
   orbit_client_data::CallstackType error_type_;
 };
 
-class CallTreeView : public CallTreeNode {
+class CallTreeView {
  public:
   [[nodiscard]] static std::unique_ptr<CallTreeView> CreateTopDownViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
 
   [[nodiscard]] static std::unique_ptr<CallTreeView>
   CreateBottomUpViewFromPostProcessedSamplingData(
       const orbit_client_data::PostProcessedSamplingData& post_processed_sampling_data,
-      const orbit_client_data::ModuleManager& module_manager,
-      const orbit_client_data::CaptureData& capture_data);
+      const orbit_client_data::ModuleManager* module_manager,
+      const orbit_client_data::CaptureData* capture_data);
+
+  CallTreeView(const orbit_client_data::ModuleManager* module_manager,
+               const orbit_client_data::CaptureData* capture_data)
+      : module_manager_{module_manager}, capture_data_{capture_data} {}
 
-  CallTreeView() : CallTreeNode{nullptr} {}
+  [[nodiscard]] CallTreeNode* GetCallTreeRootNode() { return &call_tree_root_; }
```

Comment:
`GetCallTreeRootNode` is used to return a pointer to the root so that the root can be set. I don't like this. The constructor should take the root as well, possibly as a `unique_ptr`. This would also make `CallTreeView` very lightweight, meaning it can be moved around by value and doesn't need to be stored in a smart pointer as it is now.

Reply:
Done.